### PR TITLE
feat(aws-s3-proxy): Add s3 proxy CI files

### PR DESCRIPTION
**Pull request by brandonrjacobs**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  only:
+    refs:
+      - coreweave
+  dependencies:
+    - build:release
+  variables:
+    IMAGE_NAME: {{.}}
+    REF_IMAGE: $CI_REGISTRY_IMAGE/$IMAGE_NAME:$CI_COMMIT_REF_SLUG
+    FIXED_IMAGE: $CI_REGISTRY_IMAGE/$IMAGE_NAME:$CI_COMMIT_SHORT_SHA
+    DOCKERFILE: {{$ciImagesDir}}/{{.}}/Dockerfile
```
Comment (YitzyD) [inline .gitlab-ci.yml]: What's going on here with the {{.}}

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    key: bin-${CI_COMMIT_REF_SLUG}
+    paths:
+      - bin/
+  script:
+    - go test ./... -coverprofile cover.out
+
+build:image:
+  stage: build
+  variables:
+    DOCKER_IMG: ${CONTROLLER_IMAGE}
```
Comment (YitzyD) [inline .gitlab-ci.yml]: Is `CONTROLLER_IMAGE` set in CI args?

**Review by YitzyD — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (brandonrjacobs) [reply .gitlab-ci.yml]: ah that is old i need to remove one sec.

File: .gitlab-ci.yml (hunk 1 above)
Comment (brandonrjacobs) [reply .gitlab-ci.yml]: that was from utility images, copy paste error.

**Review by YitzyD — APPROVED** [review]

**Pull request merged**